packages/flutter_test/lib/src/mock_event_channel.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+/// A mock stream handler for an [EventChannel] that mimics the native
+/// StreamHandler API.
+abstract class MockStreamHandler {
+  /// Handler for the listen event.
+  void onListen(dynamic arguments, MockStreamHandlerEventSink events);
+
+  /// Handler for the cancel event.
+  void onCancel(dynamic arguments);
+}
+
+/// Convenience class for creating a [MockStreamHandler] inline.
+class InlineMockStreamHandler extends MockStreamHandler {
+  /// Create a new [InlineMockStreamHandler] with the given [onListen] and
+  /// [onCancel] handlers.
+  InlineMockStreamHandler({
+    required void Function(dynamic arguments, MockStreamHandlerEventSink events) onListen,
+    void Function(dynamic arguments)? onCancel,
+  })  : _onListenInline = onListen,
+        _onCancelInline = onCancel;
+
+  final void Function(dynamic arguments, MockStreamHandlerEventSink events) _onListenInline;
+  final void Function(dynamic arguments)? _onCancelInline;
+
+  @override
+  void onListen(dynamic arguments, MockStreamHandlerEventSink events) => _onListenInline(arguments, events);
+
+  @override
+  void onCancel(dynamic arguments) => _onCancelInline?.call(arguments);
+}
+
+/// A mock event sink for a [MockStreamHandler] that mimics the native
+/// EventSink API.
+class MockStreamHandlerEventSink {
+  /// Create a new [MockStreamHandlerEventSink] with the given [_sink].
+  MockStreamHandlerEventSink(this._sink);
```

Comment:
nit: Don't use `this` with private variables on a public API and don't reference private variables in public docs.

Reply:
I want to make this constructor private, but that would require moving these classes to `test_default_binary_messenger.dart` which seems weird. Let me know what you think.

Reply:
You could make it a `part` to the `test_default_binary_messenger`? For all intents and purposes, that means the compiler acts like it's in that same file/library, but you can still have it organized into separate files.

Reply:
I thought `part`/`part of` was deprecated or something?

Reply:
I went ahead and just used `part`/`part of`. Let me know if we want to do it a different way.

Reply:
Should I undo the change then?

Reply:
Done